mllib/src/main/scala/org/apache/spark/mllib/tree/impurity/Entropy.scala

```diff
@@ -83,23 +83,29 @@ object Entropy extends Impurity {
  * @param numClasses  Number of classes for label.
  */
 private[spark] class EntropyAggregator(numClasses: Int)
-  extends ImpurityAggregator(numClasses) with Serializable {
+  extends ImpurityAggregator(numClasses + 1) with Serializable {
```

Comment:
sorry, trying to follow this part of the code, why do we pass (numClasses + 1) to the impurityAggregator?